### ISSUE TITLE
Finish merged layer/filter

### ISSUE THREAD

**Issue by NealHumphrey** [codefordc/housing-insights#392]
The demo-v2-options branch has our new map view UI, which includes a merged layer/zone navigation for the data choices UI. The UI of this is mostly working but not all of the data is actually being applied to the filter view. 

- [x] Get the new combined 'location' selection to apply appropriate filters to the location
- [x] Changing location type will need to clear any selected 'location' filters (since previously selected choices are no longer relevant). Need to make this obvious to user -maybe use a tooltip to show that it has been cleared? Include a 'don't show again' button that removes the tooltip for the current session (this is much easier to implement than adding cookies which can be a long term add, if we keep this ui element). 
- [x] Merge filterView.components with mapView.initialOverlays for a unified list to be used for all views. After #388 is complete, we can move a lot of this out to meta.json - for now it can be either in code or in a static config file.
- [x] For now, flag any 'layer' items that do not work with filters yet - after #388 is complete we will have the necessary data available. 
- [x] Add the legend for layers when a layer-relevant data source is selected. Move this out of the data dropdown where it was in 'layers' and instead put it in the bottom right of the map.

**Comment by NealHumphrey** [comment]
I'll be working on the first bullet points of this one today so we can close out this issue.

**Comment by NealHumphrey** [comment]
Closing this - #443 resolves the first two bullets.
